packages/fhir-converter/src/templates/cda/Sections/Encounters.hbs

```diff
@@ -1,10 +1,12 @@
-{{#if (contains (toString (toJsonString msg)) '2.16.840.1.113883.10.20.22.2.22.1')}}
-    {{#with (getFirstCdaSectionsByTemplateId msg '2.16.840.1.113883.10.20.22.2.22.1')}}
-            {{#each (toArray this.[2_16_840_1_113883_10_20_22_2_22_1].entry) as |encounterEntry|}}
-                {{>Resources/Encounter.hbs encounter=encounterEntry.encounter ID=(generateUUID (toJsonString encounterEntry))}},
-                {{#with (evaluate 'Utils/GeneratePatientId.hbs' obj=@metriportPatientId) as |patientId|}}
-                	{{>References/Encounter/subject.hbs ID=(generateUUID (toJsonString encounterEntry)) REF=(concat 'Patient/' patientId.Id)}},
-                {{/with}}
+{{#if (contains (toString (toJsonString msg)) '2.16.840.1.113883.10.20.22.2.22')}}
+    {{#with (getFirstCdaSectionsByTemplateId msg '2.16.840.1.113883.10.20.22.2.22.1' '2.16.840.1.113883.10.20.22.2.22') as |encounters|}}
```

Comment:
We were missing encounters because some were under: '2.16.840.1.113883.10.20.22.2.22' and not at all under '2.16.840.1.113883.10.20.22.2.22.1'

Reply:
So this is a really important find @jonahkaye - here is the full story:

See this immunization template for reference: https://github.com/metriport/metriport/blob/572208c0c20c90cf80f3594aabbb2fffba68f916/packages/fhir-converter/src/templates/cda/Sections/Immunization.hbs#L1-L2

Also, see list of C-CDA section templates for reference: https://build.fhir.org/ig/HL7/cda-ccda-2.2/artifacts.html#2

Note that it checks the "higher-level" section ID, and then it gets the sections matching 2 different IDs. The "higher-level" ID is the "entries optional" ID, and the other one is the "entries required" ID:
- https://build.fhir.org/ig/HL7/cda-ccda-2.2/StructureDefinition-2.16.840.1.113883.10.20.22.2.2.html
- https://build.fhir.org/ig/HL7/cda-ccda-2.2/StructureDefinition-2.16.840.1.113883.10.20.22.2.2.1.html

Several C-CDA sections, including the Encounters one you uncovered follow this pattern - let's sweep through and make sure there's no other sections missing like this.

Reply:
So my only concern here was that if we used getFirstCdaTemplate function, is their a chance their are multiple sections, one with the required id and one with the optional id?

We could use getAllCDATemplate sections, but then will we get duplicates for the sections that have both the optional and required id?

Reply:
assuming this isn't a concern any more since we check for exact match?

Reply:
Not a concern because I ran a few scripts to test and didnt find any instances of optionalId and requiredId in the same file in **different** sections. They either are both in section, or only one is.

Reply:
Tests show we arent losing data too